RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -366,6 +366,16 @@ class PumpOpsSynchronous {
                             events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), atIndex: 0)
                         }
                     }
+                    
+                    if let alarm = event as? PumpAlarmPumpEvent {
+                        switch alarm.alarmType {
+                        case .BatteryDepleted, .BatteryOutLimitExceeded, .DeviceReset:
+                            print("Found clock loss in pump history.  Ending history fetch.")
```

Comment:
That's a really good point. Ideally these stateful parses are kept short, but that case could still happen. 

> On Jul 29, 2016, at 11:26 AM, Pete Schwamb <notifications@github.com> wrote:
> 
> In RileyLinkKit/PumpOpsSynchronous.swift:
> 
> > @@ -366,6 +366,16 @@ class PumpOpsSynchronous {
> >                              events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), atIndex: 0)
> >                          }
> >                      }
> > +                    
> > +                    if let alarm = event as? PumpAlarmPumpEvent {
> > +                        switch alarm.alarmType {
> > +                        case .BatteryDepleted, .BatteryOutLimitExceeded, .DeviceReset:
> > +                            print("Found clock loss in pump history.  Ending history fetch.")
> But what I'm wondering about is this situation: there are two valid time change events after the battery out event: one to recover from the battery loss, and one that is caused by traveling, or daylight savings, or something else valid.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>